main.py

```diff
@@ -0,0 +1,364 @@
+import requests
+import json
+import datetime
+import os
+from operator import itemgetter
+
+
+# GLOBALS
+PATH = os.path.dirname(os.path.realpath(__file__))   # Gets the absolute path for the running file
+RAW_PATH = PATH + "/data/{}/raw/"
+EDIT_PATH = PATH + "/data/{}/edited/"
+FILE_NAME = "Riven_data_{}_{}.json"
+TOTAL_FILE_PATH = EDIT_PATH + "TOTAL_{}.json"
+L_MONDAY = ""
+PLATFORMS = ["PC", "PS4", "XB1", "SWI"]
+URL = "https://n9e5v4d8.ssl.hwcdn.net/repos/weeklyRivens{}.json"
+
+
+def last_monday():
+    '''Gets the last monday date'''
+    today = datetime.date.today()
+    monday = today - datetime.timedelta(days=today.weekday())
+    return monday
+
+
+def get_riven_json(platform):
+    '''Gets the json for a given platform'''
+    print(f"Checking if this week json for {platform} is saved!")
+    filename = FILE_NAME.format(platform, L_MONDAY.strftime("%y-%m-%d"))
+    file_path = f"{RAW_PATH.format(platform)}{filename}"
+
+    # Checks if the file already exists
+    if not os.path.isfile(file_path):
+        print("No file found, creating it!\n-----------------")
+        r = requests.get(URL.format(platform))
+        r.raise_for_status()
+        data = r.json()
+        with open(file_path, "w") as outfile:
+            json.dump(data, outfile, indent=4)
+    else:
+        print("File already exists!\n-----------------")
+
+
+def process_data(platform):
+    '''Calls all of the processing functions for every raw file'''
+
+    # Goes over every file in the raw folder and processes the data for it
+    for filename in os.listdir(RAW_PATH.format(platform)):
+
+        print(f"\nProcessing {filename}\n")
+
+        date = filename.rstrip(".json")[-8:]
+        raw_file_path = f"{RAW_PATH.format(platform)}{filename}"
+        edit_file_path = f"{EDIT_PATH.format(platform)}{filename}"
+
+        with open(raw_file_path, "r") as raw:
+            raw_data = json.load(raw)
+            sales = sale_calc(raw_data)
+            raw_dict = create_dict(raw_data, 1)
+            del raw_data
+
+        # Checks if the raw data is in the TOTAL file and returns the total dict
+        total_dict = total_json(platform, raw_dict, date, sales)
+
+        if os.path.isfile(edit_file_path):
```

Comment:
That's a old comment that I didn't change. `total_json` now checks if the given weeks raw data is already in the file. If you want to open the raw file only if it will be needed (for the total file or the edited one) I would need to move that check from `total_json` to before line 56 which would also require to move the check if the total file even exists. I could do all that I just don't think it matters.

Reply:
Aye understood 👌